src/variety/vslither.js

```diff
@@ -0,0 +1,420 @@
+	}
+})(["vslither"], {
+	//---------------------------------------------------------
+	// マウス入力系
+	MouseEvent: {
+		inputModes: {
+			edit: ["number", "clear", "info-line"],
+			play: [
+				"line",
+				"peke",
+				"dot",
+				"bgcolor",
+				"bgcolor1",
+				"bgcolor2",
+				"clear",
+				"info-line"
+			]
+		},
+		mouseinput_auto: function() {
+			var puzzle = this.puzzle;
+			// if (puzzle.playmode) {
+			// 	if (this.checkInputBGcolor()) {
+			// 		this.inputBGcolor();
+			// 	} else if (this.btn === "left") {
+			// 		if (this.mousestart || this.mousemove) {
+			// 			this.inputLine();
+			// 		} else if (this.mouseend && this.notInputted()) {
+			// 			this.prevPos.reset();
+			// 			this.inputpeke();
+			// 		}
+			// 	} else if (this.btn === "right") {
+			// 		if (this.mousestart || this.mousemove) {
+			// 			this.inputpeke();
+			// 		}
+			// 	}
+			// } else if (puzzle.editmode) {
+			// 	if (this.mousestart) {
+			// 		this.inputqnum();
+			// 	}
+			// }
+			if (puzzle.playmode) {
+				if (this.checkInputBGcolor()) {
+					this.inputBGcolor();
+				} else if (this.mousestart || this.mousemove) {
+					if (this.btn === "left") {
+						this.inputLine();
+					} else if (this.btn === "right") {
+						this.inputpeke();
+					}
+				} else if (this.mouseend && this.notInputted()) {
+					var cell = this.getcell();
+					if (!this.firstCell.isnull && cell !== this.firstCell) {
+						return;
+					}
+					this.prevPos.reset();
+					var cross = this.getpos(0.25).getx();
+					if (!cross.isnull) {
+						this.inputdot();
+					} else {
+						this.inputpeke();
+					}
+				}
+			} else if (puzzle.editmode) {
+				if (this.mousestart) {
+					this.inputqnum();
+				}
+			}
+		},
+
+		checkInputBGcolor: function() {
+			var inputbg = this.puzzle.execConfig("bgcolor");
+			if (inputbg) {
+				if (this.mousestart) {
+					inputbg = this.getpos(0.25).oncell();
+				} else if (this.mousemove) {
+					inputbg = this.inputData >= 10;
+				} else {
+					inputbg = false;
+				}
+			}
+			return inputbg;
+		},
+
+		mouseinput_other: function() {
+			if (this.inputMode === "dot") {
+				this.inputdot();
+			}
+		},
+		inputdot: function() {
+			var pos = this.getpos(0.25);
+			if (this.prevPos.equals(pos)) {
+				return;
+			}
+
+			var dot = pos.getDot();
+			this.prevPos = pos;
+			if (dot === null) {
+				return;
+			}
+
+			if (this.inputData === null) {
+				if (this.btn === "left") {
+					this.inputData = { 0: 1, 1: 2, 2: 0 }[dot.getDot()];
+				} else if (this.btn === "right") {
+					this.inputData = { 0: 2, 1: 0, 2: 1 }[dot.getDot()];
+				} else {
+					return;
+				}
+			}
+			dot.setDot(this.inputData);
+			dot.draw();
+		}
+	},
+
+	//---------------------------------------------------------
+	// キーボード入力系
+	KeyEvent: {
+		enablemake: true
+	},
+
+	//---------------------------------------------------------
+	// 盤面管理系
+	Cell: {
+		maxnum: 4,
+		minnum: 0,
+
+		getdir4BorderVertex1: function() {
```

Comment:
It should be possible to write this function in a much shorter way. A vertex (or Cross object) has a property `lcnt` which keeps track of how many lines are connected to it.

(this also applies to `getdir4BorderVertex1` in `slither.js`)

Reply:
Updated this, now a single line. `getdir4BorderLine1` in `slither.js` could probably be shortened in a similar way if desired, although I don't want to accidentally break it if there's something I've missed